e2e/test/scenarios/question/reproductions/22230-query-builder-aggregation-filtering.cy.spec.js

```diff
@@ -0,0 +1,43 @@
+import {
+  restore,
+  popover,
+  summarize,
+  openPeopleTable,
+  visualize,
+} from "e2e/support/helpers";
+
+describe("issue 22230", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow filtering on aggregated fields", () => {
+    openPeopleTable({ mode: "notebook" });
+
+    summarize({ mode: "notebook" });
+    popover().findByRole("option", { name: "Maximum of ..." }).click();
+    // I don't know why this can't be chained, I tried it and it didn't work.
+    popover().findByRole("option", { name: "Name" }).click();
+
+    // findByRole doesn't work here, and I don't know why.
+    cy.findByText("Pick a column to group by").click();
+    popover().findByRole("option", { name: "Source" }).click();
+
+    // can't use `filter()` because it found multiple filter buttons and Cypress only allows clicking on one element at a time.
+    cy.button("Filter").click();
+    popover().findByRole("option", { name: "Max of Name" }).click();
+    // can't add aria-label to the input, since it's past the translation period.
+    cy.findByTestId("select-button").click();
```

Comment:
I've tried to use `aria-label` for the `<SelectButton />`, but couldn't find any suitable translation to go with the label, so I used the `test-id` instead.
https://github.com/metabase/metabase/blob/17997e44166921c8f895e460eb2fe6b07612dbbf/frontend/src/metabase/core/components/SelectButton/SelectButton.tsx#L67